src/routes/comments.py

```diff
@@ -80,3 +80,127 @@ async def create_comment(
         photo_id=new_comment.photo_id,
         created_by=new_comment.created_by,
     )
+
+@router.get('/{photo_id}/comments', response_model=list[CommentResponse])
+async def get_comments(
+    photo_id: int, 
+    limit: int = Query(10, ge=10, le=500), 
+    offset: int = Query(0, ge=0), 
+    db: Session = Depends(get_db),
+    ):
+
+    """
+    The get_comments function returns a list of comments for the specified photo.
+     
+    :param photo_id: int: Specify the photo id of the comments to be returned
+    :type photo_id: int
+    :param limit: int: Limit the number of comments returned
+    :type limit: int
+    :param ge: Specify the minimum value for a parameter
+    :type ge: int
+    :param le: Specify the maximum value for a parameter
+    :type le: int
+    :param offset: int: Specify the number of comments to skip
+    :type offset: int
+    :param ge: Check if the limit parameter is greater than or equal to 10
+    :type ge: int
+    :param db: Session: Get the database session
+    :type db: Session
+    :param : Specify the number of comments to be returned
+    :return: A list of comments
+    :rtype: list[CommentResponse]
+
+    """
+    comments = await repository_comments.get_comments(photo_id, limit, offset, db)
+    if comments:
+        result = []
+        for comment in comments:
+            result.append(
+                CommentResponse(
+                    id=comment.id,
+                    comment=comment.comment,
+                    created_at= comment.created_at,
+                    updated_at= comment.updated_at,
+                    photo_id= comment.photo_id,
+                    created_by= comment.created_by,
+                    )
+                )         
+    else:
+        #checks offset parameter 
+        offset = 0
+        comments = await repository_comments.get_comments(photo_id, limit, offset, db)
+        if comments:
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail=f"Number of comments for the photo {photo_id} is less than specified in offset parameter.",
+        )
+        else: 
+            raise HTTPException(
+                status_code=status.HTTP_404_NOT_FOUND,
+                detail=f"The photo {photo_id} does not exist.",
+            )
+    return result
+
+
+@router.patch('/{photo_id}/comments')
```

Comment:
We agreed to not push extra parts. I can not verify it now.

Reply:
removed